src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java

```diff
@@ -0,0 +1,385 @@
+                                                                                  =
+      PropertyEnum.<RackType>create("variant", RackType.class);
+    public static final int                    DEFAULT_META = RackType.DEFAULT.getMetadata();
+    public static final int                    FULL_META    = RackType.FULL.getMetadata();
+
+    /**
+     * The position it faces.
+     */
+    public static final PropertyDirection FACING = BlockHorizontal.FACING;
+
+    /**
+     * The hardness this block has.
+     */
+    private static final float BLOCK_HARDNESS = 10.0F;
+
+    /**
+     * This blocks name.
+     */
+    private static final String BLOCK_NAME = "blockMinecoloniesRack";
+
+    /**
+     * The resistance this block has.
+     */
+    private static final float RESISTANCE = Float.POSITIVE_INFINITY;
+
+    /**
+     * How much light goes through the block.
+     */
+    private static final int LIGHT_OPACITY = 0;
+
+    public BlockMinecoloniesRack()
+    {
+        super(Material.WOOD);
+        initBlock();
+    }
+
+    /**
+     * initialize the block
+     * sets the creative tab, as well as the resistance and the hardness.
+     */
+    private void initBlock()
+    {
+        setRegistryName(BLOCK_NAME);
+        setUnlocalizedName(String.format("%s.%s", Constants.MOD_ID.toLowerCase(), BLOCK_NAME));
+        setCreativeTab(ModCreativeTabs.MINECOLONIES);
+        this.setDefaultState(this.blockState.getBaseState().withProperty(FACING, EnumFacing.NORTH).withProperty(VARIANT, RackType.DEFAULT));
+        setHardness(BLOCK_HARDNESS);
+        setResistance(RESISTANCE);
+        setLightOpacity(LIGHT_OPACITY);
+    }
+
+    /**
+     * Check if a certain block should be replaced with a rack.
+     *
+     * @param block the block to check.
+     * @return true if so.
+     */
+    public static boolean shouldBlockBeReplacedWithRack(final Block block)
+    {
+        return block == Blocks.CHEST || block == ModBlocks.blockRack;
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @Override
+    @Deprecated
+    public boolean isFullBlock(final IBlockState state)
+    {
+        return false;
+    }
+
+    /**
+     * @deprecated but we still need this because there is nothing better.
+     */
+    @Deprecated
+    @Override
+    public IBlockState getStateFromMeta(final int meta)
+    {
+        final EnumFacing enumFacing = EnumFacing.getHorizontal(meta);
+        return this.getDefaultState().withProperty(FACING, enumFacing);
+    }
+
+    @Override
+    public int getMetaFromState(@NotNull final IBlockState state)
+    {
+        return state.getValue(FACING).getHorizontalIndex();
+    }
+
+    /**
+     * @deprecated but we still need this because there is nothing better.
+     */
+    @Deprecated
+    @Override
+    public IBlockState getActualState(final IBlockState state, final IBlockAccess worldIn, final BlockPos pos)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 7, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)